x-pack/plugins/cases/public/containers/use_get_case.tsx

```diff
@@ -5,125 +5,34 @@
-        resolveAliasId: action.payload.aliasTargetId,
-        resolveAliasPurpose: action.payload.aliasPurpose,
-      };
-    case 'FETCH_FAILURE':
-      return {
-        ...state,
-        isLoading: false,
-        isError: true,
-      };
-    case 'UPDATE_CASE':
-      return {
-        ...state,
-        data: action.payload,
-      };
-    default:
-      return state;
-  }
-};
-
-export interface UseGetCase extends CaseState {
-  /**
-   * @param [silent] When set to `true`, the `isLoading` property will not be set to `true`
-   * while doing the API call
-   */
-  fetchCase: (silent?: boolean) => Promise<void>;
-  updateCase: (newCase: Case) => void;
-}
-
-export const useGetCase = (caseId: string): UseGetCase => {
-  const [state, dispatch] = useReducer(dataFetchReducer, {
-    isLoading: false,
-    isError: false,
-    data: null,
-    resolveOutcome: null,
-  });
+export const useGetCase = (caseId: string) => {
   const toasts = useToasts();
-  const isCancelledRef = useRef(false);
-  const abortCtrlRef = useRef(new AbortController());
-
-  const updateCase = useCallback((newCase: Case) => {
-    dispatch({ type: 'UPDATE_CASE', payload: newCase });
-  }, []);
-
-  const callFetch = useCallback(
-    async (silent: boolean = false) => {
-      try {
-        isCancelledRef.current = false;
-        abortCtrlRef.current.abort();
-        abortCtrlRef.current = new AbortController();
-        dispatch({ type: 'FETCH_INIT', payload: { silent } });
-
-        const response: ResolvedCase = await resolveCase(caseId, true, abortCtrlRef.current.signal);
-
-        if (!isCancelledRef.current) {
-          dispatch({ type: 'FETCH_SUCCESS', payload: response });
-        }
-      } catch (error) {
-        if (!isCancelledRef.current) {
-          if (error.name !== 'AbortError') {
-            toasts.addError(
-              error.body && error.body.message ? new Error(error.body.message) : error,
-              { title: i18n.ERROR_TITLE }
-            );
-          }
-          dispatch({ type: 'FETCH_FAILURE' });
-        }
-      }
+  return useQuery<ResolvedCase, ServerError>(
+    ['case', caseId],
+    () => {
+      const abortCtrlRef = new AbortController();
+      return resolveCase(caseId, true, abortCtrlRef.signal);
     },
-    [caseId, toasts]
+    {
+      onError: (error: ServerError) => {
+        if (error.name !== 'AbortError') {
+          toasts.addError(
+            error.body && error.body.message ? new Error(error.body.message) : error,
+            {
+              title: i18n.ERROR_TITLE,
+            }
+          );
+        }
+      },
+    }
   );
-
-  useEffect(() => {
-    callFetch();
-
-    return () => {
-      isCancelledRef.current = true;
-      abortCtrlRef.current.abort();
-    };
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [caseId]);
-  return { ...state, fetchCase: callFetch, updateCase };
 };
+
+export type UseGetCase = ReturnType<typeof useGetCase>;
```

Comment:
nit: You can use `UseQueryResult<ResolvedCase, ServerError>`

Reply:
I actually think this type is not necessary at all and I will probably remove it later.